google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystem.java

```diff
@@ -61,10 +62,10 @@
   public static final int BLOCK_SIZE_DEFAULT = 2 * 1024 * 1024;
   public static final FileTime FILE_TIME_UNKNOWN = FileTime.fromMillis(0);
   public static final Set<String> SUPPORTED_VIEWS = ImmutableSet.of(BASIC_VIEW, GCS_VIEW);
-
   private final CloudStorageFileSystemProvider provider;
   private final String bucket;
   private final CloudStorageConfiguration config;
+  private static final Map<String, CloudStorageFileSystem> FILE_SYSTEM_MAP = new HashMap<>();
```

Comment:
Sorry, I gave you a bit wrong advice. The map should `Map<String, Set<CloudStorageFileSystem>>`